File: api/src/Kernel/Domain/Billing/Entities/SubscriptionPlan.cs
using System.Collections.Generic;

public class SubscriptionPlan : Entity<SubscriptionPlan> {
    public string Name { get; set; } = null!;
    public string ExternalId { get; set; } = null!;
    public List<SubscriptionPlanPrice> Prices { get; set; } = new List<SubscriptionPlanPrice>();
}

File: app/src/modules/admin/api/entities/subscription-plan.ts
import { SubscriptionPlanPrice } from '@/modules/admin/api/entities/subscription-plan-price';
import { Entity } from '@/core';

export class SubscriptionPlan extends Entity {
    constructor(public name: string, public prices: SubscriptionPlanPrice[], public roleId: string | null = null) {
        super();
    }
}
